### PR TITLE
Document the usage of timm as backbone provider

### LINKED ISSUE #543: Backbone/model weights source should be more obvious to figure out.
I saw `torchvision` in the requirements and immediately assumed that the pre-trained model weights would come from there (I didn't know `timm` until very recently). 

I saw they are actually coming from `timm` and realized it's not a well documented information. 

I have two suggestions: 

1) (easy) mention it in the `readme.md` and add it to the docs.

2) (maybe easy?) make the pre-trained weights source a parameter that would show up in the config file (it could come from `torchvision` for instance)

### PR DESCRIPTION
**Pull request by jpcbertoldo**
# Description

Add some text in the documentation to make it clearer that timm is used to fetch pre-trained models.

- Fixes #543

## Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which refactors the code base)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [X] Documentation improvement

## Checklist

- [ ] My code follows the [pre-commit style and check guidelines](https://openvinotoolkit.github.io/anomalib/guides/using_pre_commit.html#pre-commit-hooks) of this project.
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [X] N/A

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -129,6 +129,42 @@ optimization:
 ```
 
 ## Custom Dataset
+# Feature extraction & (pre-trained) backbones
```
Comment (djdameln) [inline README.md:132]: The formatting is a bit off here. I think this header
`## Custom Dataset`
should be placed under your section, and
`# Feature extraction & (pre-trained) backbones`
should be 
`## Feature extraction & (pre-trained) backbones`

File: docs/source/models.rst [hunk 2]
```diff
@@ -31,6 +31,45 @@ Loading Models
...
 
+
+Feature extraction & (pre-trained) backbones
+---------------------------------------------
+
+Several models will use a pre-trained model to extract feature maps from its internal submodules -- the *backbone*.
+
+All the pre-trained backbones come from the package  `PyTorch Image Models (timm) <https://github.com/rwightman/pytorch-image-models>`_ and are wrapped by the class FeatureExtractor.
+
+For an introductiont to timm, please check the `Getting Started with PyTorch Image Models (timm): A Practitioner’s Guide <https://towardsdatascience.com/getting-started-with-pytorch-image-models-timm-a-practitioners-guide-4e77b4bf9055>`_, in particular the introduction about models and the section about `feature extraction <https://towardsdatascience.com/getting-started-with-pytorch-image-models-timm-a-practitioners-guide-4e77b4bf9055#b83b:~:text=ready%20to%20train!-,Feature%20Extraction,-timm%20models%20also>`_.
```
Comment (ashwinvaidya17) [inline docs/source/models.rst:42]: Typo `introductiont` -> `introduction`

**Review by ashwinvaidya17 — CHANGES_REQUESTED** [review]
Thanks for updating the documentation! I have a minor comment.

**Review by jpcbertoldo — COMMENTED** [review]
requested changes done

**Comment by ashwinvaidya17** [comment]
@jpcbertoldo can you have a look at the issues raised by codacy?

**Comment by jpcbertoldo** [comment]
I'll try if i have time today.
Btw tests were failing in my local env.
I think it was in the other branch but it wasn't related to my changes. 
I was wondering if we could arrange a call so i can get a hand with that.

**Comment by ashwinvaidya17** [comment]
> I'll try if i have time today. Btw tests were failing in my local env. I think it was in the other branch but it wasn't related to my changes. I was wondering if we could arrange a call so i can get a hand with that.

Sure take your time. I just ran the CI and I can see that it fails for linting checks but we can have a call in case `tox` is not working on your machine. Maybe we might have to update the documentation. Let's continue that discussion in an email thread.

**Comment by jpcbertoldo** [comment]
this was integrated in #576

**Comment by watertianyi** [comment]
I used convnextv2_huge.fcmae_ft_in22k_in1k_512 to replace the pre-trained network wide_resnet50_2 in patchcore, and used layers=['stages.1', 'stages.2'] for classification. The indicator effect is far worse than wide_resnet50_2. Isn't convnextv2 better than wide_resnet50_2? Why Would it be so bad?

**Pull request closed without merging**